glass-easel/src/shadow_root.ts

```diff
@@ -399,307 +398,333 @@ export class ShadowRoot extends VirtualNode {
    * Iterate slots
    */
   forEachSlot(f: (slot: Element) => boolean | void) {
-    if (this._$singleSlot) {
-      f(this._$singleSlot)
-    } else if (this._$slots) {
-      const slots = Object.values(this._$slots)
+    const slotMode = this._$slotMode
+    if (slotMode === SlotMode.Direct) {
+      throw new Error('Cannot iterate slots in directSlots')
+    }
+    if (slotMode === SlotMode.Single) {
+      if (this._$singleSlot) f(this._$singleSlot)
+    } else if (slotMode === SlotMode.Multiple) {
+      const slots = Object.values(this._$slots!)
       for (let i = 0; i < slots.length; i += 1) {
         if (f(slots[i]!) === false) break
       }
-    } else if (this._$dynamicSlotSet) {
-      const iterator = this._$dynamicSlotSet.values()
-      for (let next = iterator.next(); !next.done; next = iterator.next()) {
-        if (f(next.value) === false) break
+    } else if (slotMode === SlotMode.Dynamic) {
+      for (let it = this._$subtreeSlotStart; it; it = it.next) {
+        if (f(it.value) === false) break
       }
     }
   }
 
   /**
-   * Iterate elements with their slots
+   * Iterate elements with their slots (slots-inherited nodes included)
```

Comment:
Add return-value document for following several fn

Reply:
> Add return-value document for following several fn

Added